tests/integration/targets/ec2_vpc_nat_gateway/tasks/main.yml

```
@@ -145,6 +145,35 @@
  #     that:
  #       - create_ngw.changed
 
+
+  # ============================================================
+  - name: Get NAT gateway with specific filters (state and subnet)
+    ec2_vpc_nat_gateway_info:
+      filters:
+        subnet-id: "{{ subnet_id }}"
+        state: ['available']
+    register: avalaible_ngws
+
+  - name: Assert success
+    assert:
+      that:
+          - avalaible_ngws is successful
+          - avalaible_ngws.result | length == 1
```

It'd be good to assert the shape of results too:

```suggestion
    assert:
      that:
          - avalaible_ngws is successful
          - avalaible_ngws.result | length == 1
          - '"create_time" in first_ngw'
          - '"nat_gateway_addresses" in first_ngw'
          - '"nat_gateway_id" in first_ngw'
          - first_ngw.nat_gateway_id == create_ngw.nat_gateway_id
          - '"state" in first_ngw'
          - first_ngw.state == 'available'
          - '"subnet_id" in first_ngw'
          - first_ngw.subnet_id == subnet_id
          - '"tags" in first_ngw'
          - '"vpc_id" in first_ngw'
          - first_ngw.vpc_id == vpc_id
    vars:
      first_ngw: '{{ avalaible_ngws.result[0] }}'
```

(I'm not a big fan of the lazy 'The result of the describe, converted to ansible snake case style.' results documentation either)

---

Hi @tremble and @jillr, thank you for reviewing this. I tried to add more assertions even for the other tasks. Let me know please if this makes sense for you or if there is something else I can do to improve the code. Thank you.

---

@jillr, if the changes I've applied here make sense to you, I'll try to do the same for the internet gateway as well.

---

LGTM - thanks